gadgets/ci/image_inspect/test/unit/image_inspect_test.go

```diff
@@ -0,0 +1,260 @@
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/runtime/local"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/gadgetrunner"
+
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/environment/local"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/traceloop/tracer"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/ebpf"
+	_ "github.com/inspektor-gadget/inspektor-gadget/pkg/operators/wasm"
+)
+
+type extraInfoMap struct {
+	Name string
+	Type string
+}
+
+type extraInfoProgram struct {
+	Section string
+	Source  string
+}
+
+type extraInfoVariable struct {
+	Name   string
+	Offset uint64
+	Size   uint64
+	Map    string
+}
+
+func TestInspectCmd(t *testing.T) {
+	gadgettesting.InitUnitTest(t)
+	t.Parallel()
+
+	info, err := extractGadgetInfo(true)
+	if err != nil {
+		t.Fatalf("Error getting gadget info: %v\n", err)
+	}
+
+	// Validate eBPF programs
+	expectedProgs := []string{"program1", "program2", "program3", "program4"}
+	execProgData := info.ExtraInfo.Data["ebpf.programs"].Content
+	var gotPrograms []extraInfoProgram
+	if err := json.Unmarshal(execProgData, &gotPrograms); err != nil {
+		t.Fatalf("unmarshal ebpf.programs: %v", err)
+	}
+	var gotProgramsNames []string
+	for _, v := range gotPrograms {
+		gotProgramsNames = append(gotProgramsNames, v.Section)
+	}
+	require.IsType(t, []extraInfoProgram{}, gotPrograms, "ebpf.programs: expected []extraInfoProgram, got %T", gotPrograms)
+	require.Len(t, gotProgramsNames, len(expectedProgs), "ebpf.programs: expected %d programs, got %d", len(expectedProgs), len(gotProgramsNames))
+	for _, prog := range expectedProgs {
+		require.Contains(t, gotProgramsNames, prog, "ebpf.programs: %s not in got list", prog)
+	}
+
+	// Validate eBPF maps
+	expectedMaps := []extraInfoMap{
+		{Name: "map1", Type: "Hash"},
+		{Name: "map2", Type: "Hash"},
+	}
+	execMapData := info.ExtraInfo.Data["ebpf.maps"].Content
+	var gotMaps []extraInfoMap
+	if err := json.Unmarshal(execMapData, &gotMaps); err != nil {
+		t.Fatalf("unmarshal ebpf.maps: %v", err)
+	}
+	sort.Slice(gotMaps, func(i, j int) bool {
+		return gotMaps[i].Name < gotMaps[j].Name
+	})
+	require.IsType(t, []extraInfoMap{}, gotMaps, "ebpf.maps: expected []extraInfoMap, got %T", gotMaps)
+	require.Equal(t, expectedMaps, gotMaps, "ebpf.maps mismatch")
+
+	// Validate eBPF variables
+	expectedVars := []string{"gadget_mapiter_qdisc___map1", "gadget_param_ifindex", "gadget_param_targ_ms", "ifindex", "targ_ms"}
+	execVarData := info.ExtraInfo.Data["ebpf.variables"].Content
+	var gotVars []extraInfoVariable
+	if err := json.Unmarshal(execVarData, &gotVars); err != nil {
+		t.Fatalf("unmarshal ebpf.variables: %v", err)
+	}
+	var gotVarNames []string
+	for _, v := range gotVars {
+		gotVarNames = append(gotVarNames, v.Name)
+	}
+	require.IsType(t, []extraInfoVariable{}, gotVars, "ebpf.variables: expected []extraInfoVariable, got %T", gotVars)
```

Comment:
This will be always true as you're defining it above. When I said we should validate the type, I meant the `ContentType` field 🙂.